src/rules/pycodestyle/rules/ambiguous_names.rs

```diff
@@ -0,0 +1,49 @@
+use crate::ast::types::Range;
+use crate::registry::Diagnostic;
+use crate::violations;
+
+fn is_ambiguous_name(name: &str) -> bool {
+    name == "l" || name == "I" || name == "O"
+}
+
+/// E741
+pub fn ambiguous_variable_name(name: &str, range: Range) -> Option<Diagnostic> {
+    if is_ambiguous_name(name) {
+        Some(Diagnostic::new(
+            violations::AmbiguousVariableName(name.to_string()),
+            range,
+        ))
+    } else {
+        None
+    }
+}
+
+/// E742
+pub fn ambiguous_class_name<F>(name: &str, locate: F) -> Option<Diagnostic>
+where
+    F: FnOnce() -> Range,
+{
+    if is_ambiguous_name(name) {
+        Some(Diagnostic::new(
+            violations::AmbiguousClassName(name.to_string()),
+            locate(),
+        ))
+    } else {
+        None
+    }
+}
+
+/// E743
+pub fn ambiguous_function_name<F>(name: &str, locate: F) -> Option<Diagnostic>
+where
+    F: FnOnce() -> Range,
+{
+    if is_ambiguous_name(name) {
+        Some(Diagnostic::new(
+            violations::AmbiguousFunctionName(name.to_string()),
+            locate(),
+        ))
+    } else {
+        None
+    }
+}
```

Comment:
I also meant to mention this, I put three different rules here but they are almost exactly the same and use the same helper method. I wasn't sure if you wanted to stick to one rule per file or if this is ok @charliermarsh

Reply:
I think it'd be _slightly_ preferable to split this into three files and reference a shared helper in `pycodestyle/helpers.rs`, if you don't mind?

Reply:
Can do!

Reply:
The reason being that we may start using the top-level rustdoc at the top of the violation file to automatically generate documentation about the rule (e.g., what does it do, why is it recommended, etc.) -- so having each rule in its own file has benefits unrelated to the code structure itself :)

Reply:
Yup totally makes sense. I've made that change!